core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -915,6 +915,17 @@ void *cuda_resize_scratch_space(std::int64_t bytes, bool force_shrink) {
   return ptr;
 }
 
+void cuda_prefetch_pointer(const Cuda &space, void *const ptr, size_t bytes,
+                           bool to_device) {
```

Comment:
Did you consider using an enum instead of a boolean?

Reply:
Ping

Reply:
what you mean with an enum?

Reply:
You mean like "toDevice" "toHost"?

Reply:
If you mean that: I am hesitant to add more to the API surface. For now this adds exactly one function inside of Kokkos::Impl. If I add enums I have to add an additional thing.

Reply:
Something like `cudaMemcpyKind` where you'd explicitly say "to device" or "to host".

Reply:
I just looked at that and none of them look like a good fit. I mean the prefetch thing doesn't really do anything if the pages are already on the right side so cudaMemcpyHostToDevice feels kind of wrong. And I definitely don't want to introduce my own new type.